janus_core/calculations/single_point.py

```diff
@@ -219,7 +220,7 @@ def set_calculator(
 
         if isinstance(self.struct, list):
             for struct in self.struct:
-                struct.calc = calculator
+                struct.calc = copy(calculator)
```

Comment:
Does this want to be a copy or a deepcopy?

Reply:
Good question, this seemed to be sufficient for the results, but perhaps a deepcopy would be safest just in case.